modules/analysis-common/src/main/java/org/opensearch/analysis/common/CommonAnalysisModulePlugin.java

```diff
@@ -187,7 +188,8 @@ public Collection<Object> createComponents(
         NodeEnvironment nodeEnvironment,
         NamedWriteableRegistry namedWriteableRegistry,
         IndexNameExpressionResolver expressionResolver,
-        Supplier<RepositoriesService> repositoriesServiceSupplier
+        Supplier<RepositoriesService> repositoriesServiceSupplier,
+        TaskResourceTrackingService taskResourceTrackingService
```

Comment:
Can we add the taskResourceTrackingService into one of the above parameters? If not this will be a breaking change and we will sadly need to update the `createComponents` function for all plugins :(